File: 01-09/Problem_3/Problem3.java
// javac Problem3.java
// java -classpath . Problem3

public class Problem3 {

    public static void main(String[] args) {
        long number = 600851475143L;
        int factor = 2;
        while (factor < number) {
            while (number%factor == 0) number /= factor;
            factor += 1;
        }
        System.out.println(number);
    }
}

File: 01-09/Problem_3/problem3.rs
// rustc problem2.rs

fn main() {
    let mut number: u64 = 600_851_475_143;
    let mut factor = 2;
    while factor < number {
        while number%factor == 0 {
            number /= factor;
        }
        factor += 1;
    }
    println!("{}", number);
}